src/main/java/edu/ucsb/cs156/gauchoride/controllers/CartController.java

```diff
@@ -0,0 +1,113 @@
+package edu.ucsb.cs156.gauchoride.controllers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.ucsb.cs156.gauchoride.entities.Cart;
+import edu.ucsb.cs156.gauchoride.repositories.CartRepository;
+
+import edu.ucsb.cs156.gauchoride.errors.EntityNotFoundException;
+
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import javax.validation.Valid;
+
+
+@Api(description = "Cart")
+@RequestMapping("/api/carts")
+@RestController
+//@Slf4j
+public class CartController extends ApiController {
+
+    @Autowired
+    CartRepository cartRepository;
+
+    @ApiOperation(value = "List all carts")
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
This should be either 
```suggestion
    @PreAuthorize("hasRole('ROLE_ADMIN')")
```

or


```suggestion
    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER')")
```

I'm ok with just `ROLE_ADMIN` for now to keep things simple.  But `ROLE_USER` is definitely incorrect.   We don't want ordinary `ROLE_USER` users to have pretty much no access to any of the CRUD operations.

Reply:
Changed to @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER')")